File: Pandacap.Data/TheirContent/ActivityPub/RemoteActivityPubReply.fs
namespace Pandacap.Data

open System
open System.ComponentModel.DataAnnotations

/// A remote ActivityPub post that is a reply to one of this app's posts (a UserPost or AddressedPost).
type RemoteActivityPubReply() =
    [<Key>]
    member val Id = Guid.Empty with get, set

    member val ObjectId = "" with get, set
    member val InReplyTo = Guid.Empty with get, set

    member val Public = false with get, set
    member val Approved = false with get, set

    member val CreatedBy = "" with get, set
    member val Username = nullString with get, set
    member val Usericon = nullString with get, set
    member val CreatedAt = DateTimeOffset.MinValue with get, set
    member val Summary = nullString with get, set
    member val Sensitive = false with get, set
    member val Name = nullString with get, set
    member val Content = nullString with get, set

File: Pandacap.Database/RemoteActivityPubReply.cs
namespace Pandacap.Database
{
    /// <summary>
    /// A remote ActivityPub post that is a reply to one of this app's posts (a UserPost or AddressedPost).
    /// </summary>
    public class RemoteActivityPubReply
    {
        public Guid Id { get; set; }

        public string ObjectId { get; set; } = "";

        public string InReplyTo { get; set; } = "";

        public string CreatedBy { get; set; } = "";

        public string? Username { get; set; }

        public string? Usericon { get; set; }

        public DateTimeOffset CreatedAt { get; set; }

        public string? Summary { get; set; }

        public bool Sensitive { get; set; }

        public string? Name { get; set; }

        public string? HtmlContent { get; set; }
    }
}
